code/modules/hydroponics/fermenting_barrel.dm

```diff
@@ -0,0 +1,99 @@
+/obj/structure/fermenting_barrel
+	name = "wooden barrel"
+	desc = "A large wooden barrel. You can ferment fruits and such inside it, or just use it to hold liquid."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "barrel"
+	density = TRUE
+	anchored = FALSE
+	container_type = DRAINABLE | AMOUNT_VISIBLE
+	var/open = FALSE
+	pressure_resistance = 2 * ONE_ATMOSPHERE
+	max_integrity = 300
+	var/list/ferment_times = list()
+
+/obj/structure/fermenting_barrel/Initialize()
+	create_reagents(500) //Half of a beer keg, since it can be refilled.
+	. = ..()
+
+/obj/structure/fermenting_barrel/Destroy()
+	QDEL_LIST(ferment_times)
+	. = ..()
+
+/obj/structure/fermenting_barrel/examine(mob/user)
+	. = ..()
+	to_chat(user, "<span class='notice'>It is currently [open?"open, letting you pour liquids in.":"closed, letting you draw liquids from the tap."]</span>")
+
+/obj/structure/fermenting_barrel/process()
+	if(contents.len == 0)
+		STOP_PROCESSING(SSobj,src)
+		return
+
+	for(var/obj/item/reagent_containers/food/snacks/grown/fruit in contents)
+		if(ferment_times["[REF(fruit)]"])
+			ferment_times["[REF(fruit)]"]--
+		else
```

Comment:
This whole proc could be replaced with individual timers for object fermentation

Reply:
I really don't know why I didn't think of that.

Reply:
Also, if you rip off the method I used for the other brewing setup, you can make arbitrary things distillable.